posts/2023-06-07-kotlin-ktor-keycloak.adoc

```diff
@@ -0,0 +1,357 @@
+= JWT token validation in Ktor with Client Credentials Flow
+LeonorLunatech
+v1.0, 2023-06-07
+:title: JWT token validation in Ktor with Client Credentials Flow
+:imagesdir: ../media/2023-06-07-kotlin-ktor-keycloak
+:lang: en
+:tags: [kotlin, ktor, keycloak]
+
+At Lunatech we build many internal tools that facilitate internal processes like keeping track of employees' vacations, share internal events and workshops and even one just for the days Lunatech offers us lunch ;) So, it's only natural that we end up creating integrations between these applications.
+
+The goal of this post is to explore the scenario where one application authenticates with another one by using the client credentials flow, in https://kotlinlang.org/[Kotlin], https://ktor.io/[Ktor] and https://www.keycloak.org/[Keycloak].
+To achieve this, we are going to build a simple personal `luna-vacation-bot` that will inform you daily, via a https://slack.com/[Slack] message, whether the present day is a vacation day or not. For this we will need to implement a dummy `luna-vacation-api` as well.
```

Comment:
I know I prefixed the projects with `luna` but in reality this system is not bounded to Lunatech, so I think that adding that it will `inform a Lunatech employee`  may give the feeling that no one else can try to run these projects locally